src/main/java/com/hackclub/hccore/PlayerData.java

```diff
@@ -63,8 +73,10 @@ public void setAfk(boolean afk) {
     this.getTeam().color(NamedTextColor.nearestTo(newColor));
     this.getTeam().suffix(Component.text(newSuffix).color(newColor));
 
-    Event event = new PlayerAFKStatusChangeEvent(this.player, afk);
-    this.player.getServer().getPluginManager().callEvent(event);
+    if (this.player != null) {
+      Event event = new PlayerAFKStatusChangeEvent(this.player, afk);
+      this.player.getServer().getPluginManager().callEvent(event);
```

Comment:
as team returns null for offline players the nullcheck should be at the beginning, and return a no-op (no afk changing while offline, as its not persisted after disconnect anyway iirc)